model/textparse/openmetricsparse.go

```diff
@@ -219,12 +226,79 @@ func (p *OpenMetricsParser) Exemplar(e *exemplar.Exemplar) bool {
 	return true
 }
 
-// CreatedTimestamp returns nil as it's not implemented yet.
-// TODO(bwplotka): https://github.com/prometheus/prometheus/issues/12980
+// CreatedTimestamp returns the created timestamp as *int64 of a given Metric.
+// It searches for the _created line of a given metric and uses an ephemeral createdTimestampParser to peek ahead and find the _created line.
 func (p *OpenMetricsParser) CreatedTimestamp() *int64 {
+	var lbs labels.Labels
+	p.Metric(&lbs)
+	lbs = lbs.DropMetricName()
+
+	newParser := createdTimestampParser(p)
+loop:
+	for {
+		switch t, _ := newParser.Next(); t {
+		case EntrySeries:
+			// TODO: potentially broken? Missing type?
+			if newParser.mName != p.mName {
+				return nil
+			}
+
+			var newLbs labels.Labels
+			newParser.Metric(&newLbs)
+			name := newLbs.Get(model.MetricNameLabel)
+			if !strings.HasSuffix(name, "_created") {
+				continue
+			}
+
+			newLbs = newLbs.DropMetricName()
+			switch p.mtype {
+			case model.MetricTypeCounter, model.MetricTypeHistogram, model.MetricTypeSummary:
+				labelDiffs := lbs.MatchLabels(false, newLbs.ExtractNames()...)
+				if labelDiffs.Len() != 0 {
+					if !labelDiffs.Contains("quantile", "le") || labelDiffs.Len() != 1 {
+						return nil
+					}
+				}
+			default:
+				// if mtype is not counter, summary or histogram, it won't have a valid _created line
+				return nil
+			}
+
+			ct := int64(newParser.val)
+			return &ct
+		default:
+			break loop
+		}
+	}
 	return nil
```

Comment:
So probably it's not smart from mentorship, but I went into rabbit hole of understanding this code and what could be improved.. and I ended rewriting a bit. Sometimes code is worth thousands words (or thousands comments on PR). 

This is tricky code, so we could spend on this a few weeks to get @Maniktherana to figure things out one by one, but I wonder.. perhaps it's time to move to something else, more features we planned here.

So I will paste here, let's see. Feel free to disagree and challenge (I didn't build or test this code so can be totally broken). Happy to walk through the code on our sync if time allows.

```suggestion
func (p *OpenMetricsParser) CreatedTimestamp() *int64 {
	if !typeRequiresCT(	p.mtype) {
		// Not a CT supported metric type, fast path.
		return nil
	}
	
	var (
		currLset                labels.Labels
		buf                     []byte
		peekWithoutNameLsetHash uint64
	)
	p.Metric(&currLset)
	currWithoutNameLsetHash, _ := currLset.HashWithoutLabels(nil, labels.MetricName)

	// Search for the _created line for the currName using ephemeral parser until
	// we see EOF or new metric family. We have to due it as we don't know where (and if)
	// that CT line is.
	// TODO(bwplotka): Make sure OM 1.1/2.0 pass CT via metadata or exemplar-like to avoid this.
	peek := createdTimestampParser(p)
	for {
		eType, err := peek.Next()
		if err != nil {
			// This means p will give error too later on, so def no CT line found.
			// This might result in partial scrape with wrong/missing CT, but only
			// spec improvement would help.
			// TODO(bwplotka): Make sure OM 1.1/2.0 pass CT via metadata or exemplar-like to avoid this.
			return nil
		}
		if eType != EntrySeries {
			// Assume we hit different family, no CT line found.
			return nil
		}

		// We are sure this series is for sure the same metric family as in currLset
		// because otherwise we would have EntryType first which we ruled out before.

		var peekedLset labels.Labels
		peek.Metric(&peekedLset)
		peekedName := peekedLset.Get(model.MetricNameLabel)
		if !strings.HasSuffix(peekedName, "_created") {
			// Not a CT line, search more.
			continue
		}
		
		// We got a CT line here, but let's search if CT line is actually for our series, edge case.
		peekWithoutNameLsetHash, buf = peekedLset.HashWithoutLabels(buf, labels.MetricName)
		if peekWithoutNameLsetHash != currWithoutNameLsetHash {
			// CT line for a different series, for our series no CT.
			return nil
		}

		ct := int64(peek.val)
		return &ct
	}
}

func typeRequiresCT(t model.MetricType) bool {
	switch t {
	case model.MetricTypeCounter, model.MetricTypeSummary, model.MetricTypeHistogram:
		return true
	default:
		return false
	}
}
```

Reply:
Trying this out rn and currently it breaks for summaries when it checks for CT with a quantile on this line: 
`if peekWithoutNameLsetHash != currWithoutNameLsetHash`

Reply:
So from what I can gather this is happening because summaries may have `quantile` labels
similarly `le` would be present in histograms but this may not (probably never) be present in in the `_created` line.

Hence the condition fails

Reply:
Yea looks like that was the issue

So I had to drop "le" or "quantile" depending on if we're on a histogram or summary

All tests pass now!